### DISCUSSION TITLE
Blade

### DISCUSSION THREAD

**Discussion by jasonvarga** [statamic/cms#5057] [General]
In 3.3, we're working to improve Blade compatibility.

You can use Blade, and it works fine for the most part, but we've heard people mention that it's just not quite as nice as Antlers.

Anyone that has worked with Blade, would you be able to share anything you struggled with, any workarounds that you had to implement, anything that just felt a bit ugly to you, or anything else!

We know that dealing with augmentation is a pain point, but we'd like some concrete examples if possible.

Thanks!

**Comment by riasvdv** [comment]
I have a few sites built with Blade and to be honest once you get through a few gotchas it's all smooth sailing

- https://github.com/edalzell/statamic-blade helps a lot, especially replacing the ViewServiceProvider to have automatic augmentation, a "Statamic-native" way to enable this would be great though.
- The `modify` and `tag` helpers I created are really useful, good to see they're being added to core (https://github.com/statamic/cms/pull/5058)
- Without automatic augmentation it's really annoying to have to check every time if a value is an instance of `Value` and then call `->value()` or is it `->augmentedValue()` or is it `->antlersValue()` or `toAugmentedArray()` on it to get the actual value(s). Then sometimes nested values are augmented, sometimes they're not. (Basically all the logic here: https://github.com/edalzell/statamic-blade/blob/main/src/Concerns/AugmentsValues.php)
- Fetching entries with `Entry::query()` is nice, but the Statamic tags give you a lot of included niceties like fetching published entries only by default, which is what lead me to implement the tag helper. The extra Blade directives in Erin's addon attempt to solve some of those cases, but it doesn't cover all of them, so you can't just translate an Antlers tag to the Blade directive directly. 
 
Maybe having a `ViewEntryQuery` that sets some defaults, that then both PHP (& Blade) and Antlers tags use could help streamline this? It could include augmentation on the `->get()` call or `->getAugmented()` at then end as well to not have to augment the results manually in PHP.

- Antlers handles `null` values and errors really well by just not displaying something when it throws an exception, it's about the only thing I miss using Blade 😄
